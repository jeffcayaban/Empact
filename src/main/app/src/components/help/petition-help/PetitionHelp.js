import React from 'react';
import {Col, Grid, ListGroup, ListGroupItem, Row} from "react-bootstrap";
import BackButton from "../../common/BackButton";

/**
 * Displays the links to help pages that are relevant to petitions.
 */

class PetitionHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Petition Help</h2>
                </div>

                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <ListGroup>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/view`}>
                                        Viewing Petitions
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/create`}>
                                        Creating a Petition
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/edit`}>
                                        Editing a Petition
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/delete`}>
                                        Deleting a Petition
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/close`}>
                                        Closing a Petition
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/petition/search`}>
                                        Searching Petitions
                                    </ListGroupItem>
                                </ListGroup>
                                <BackButton link={"/help"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default PetitionHelp;