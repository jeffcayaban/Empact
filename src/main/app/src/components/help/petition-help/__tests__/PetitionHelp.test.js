import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import PetitionHelp from "../PetitionHelp";

describe('PetitionHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<PetitionHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
