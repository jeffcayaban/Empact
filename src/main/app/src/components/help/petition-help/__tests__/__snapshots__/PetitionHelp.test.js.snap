// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Petition Help
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <ListGroup
            bsClass="list-group"
          >
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Viewing Petitions
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Creating a Petition
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Editing a Petition
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Deleting a Petition
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Closing a Petition
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Searching Petitions
            </ListGroupItem>
          </ListGroup>
          <BackButton
            link="/help"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
